bats/tests/profile/create-profile-output.bats

```diff
@@ -0,0 +1,424 @@
+            assert_failure
+            assert_output --partial $"Error: flag needs an argument: --$input"
+        done
+    done
+}
+
+too_many_input_formats() {
+    run rdctl create-profile "$@"
+    assert_failure
+    assert_output --partial 'too many input formats specified: must specify exactly one input format of "--input FILE|-", "--body|-b STRING", or "--from-settings"'
+}
+
+@test 'complains when multiple input sources are specified' {
+    for type in reg plist; do
+        too_many_input_formats --output $type --input some-file.txt -b moose
+        too_many_input_formats --output $type --input some-file.txt --from-settings
+        too_many_input_formats --output $type --input some-file.txt -b moose --from-settings
+        too_many_input_formats --output $type -b moose --from-settings
+    done
+}
+
+@test "complains when input file doesn't exist" {
+    run rdctl create-profile --output reg --input /no/such/file/here
+    assert_failure
+    assert_output --partial 'Error: open /no/such/file/here:'
+}
+
+@test 'report invalid parameters for plist' {
+    run rdctl create-profile --output=plist --from-settings --hive=fish
+    assert_failure
+    assert_output --partial $"registry hive and type can't be specified with \"plist\""
+
+    run rdctl create-profile --output plist --from-settings --type=writer
+    assert_failure
+    assert_output --partial $"registry hive and type can't be specified with \"plist\""
+}
+
+@test 'report unrecognized output-options' {
+    run rdctl create-profile --output=pickle
+    assert_failure
+    assert_output --partial 'received unrecognized "--output FORMAT" option of "pickle"; "plist" or "reg" must be specified'
+}
+
+@test 'report unrecognized registry sub-options' {
+    run rdctl create-profile --output=reg --hive=hklm --type=ruff --from-settings
+    assert_failure
+    assert_output --partial 'invalid registry type of "ruff" specified'
+}
+
+# Happy tests follow
+
+# Sample input-generating functions
+
+complex_json_data() {
+    echo '{"kubernetes": {"enabled": false}, "containerEngine": { "allowedImages": {"patterns": ["abc", "ghi", "def"] } }, "WSL": { "integrations": { "first": true, "second": false } } }'
+}
+export -f complex_json_data
+
+simple_json_data() {
+    echo '{ "kubernetes": {"version": "moose-head" }}'
+}
+export -f simple_json_data
+
+json_with_special_chars() {
+    cat <<'EOF'
+{ "application": {
+    "extensions": {
+        "allowed": {
+          "enabled": false,
+          "list": ["less-than:<", "greater:>", "and:&", "d-quote:\"", "emoji:😀"]
+        },
+        "installed": {
+            "key-with-less-than: <": true,
+            "key-with-ampersand: &": true,
+            "key-with-greater-than: >": true,
+            "key-with-emoji: 🐤": false
+        }
+    }
+  },
+  "containerEngine": {
+    "name": "small-less-<-than"
+  }
+}
+EOF
+}
+export -f json_with_special_chars
+
+assert_full_setting_registry_output() {
```

Comment:
Should all the `full_setting` names be changed to `full_settings` to match the `--from-settings` option name?

Reply:
ok